### PR TITLE
AMBARI-23456. Add mkdocs support & markdown skeletons

### PR DESCRIPTION
**Pull request by oleewere**
# What changes were proposed in this pull request?
- fix some docs
- generate output shipper configs as well
- add mkdocs support
- put javadoc + rest api doc as well to generated site
- add markdown skeletons (i will need to fill those + add nice pictures)

note: probably we will use docker for mkdocs build, right now it is expected to be installed

## How was this patch tested?
full test: `make serve-site`, it can take long because of 2 web build, if modules are already installed then its enough to run: `make javadoc`, `make prop-docs-only` then `make serve-site-only`

Please review @g-boros

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/106/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/107/
Test FAILed.
Test FAILured.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/108/
Test PASSed.

File: ambari-logsearch-config-json/src/main/java/org/apache/ambari/logsearch/config/json/model/inputconfig/impl/InputConfigImpl.java [hunk 1]
```diff
@@ -39,15 +39,17 @@
   @ShipperConfigElementDescription(
     path = "/input",
     type = "list of json objects",
-    description = "A list of input descriptions"
+    description = "A list of input descriptions",
+    examples = {"{\"input\" : [ {\"type\": \"myinput_service_type\"}] }"}
```
Comment (kasakrisz) [inline ambari-logsearch-config-json/src/main/java/org/apache/ambari/logsearch/config/json/model/inputconfig/impl/InputConfigImpl.java]: What do you think? Would adding the following is helpful:
```
where myinput_service_type is like ambari_server, hdfs_namenode etc...
```

File: ambari-logsearch-config-json/src/main/java/org/apache/ambari/logsearch/config/json/model/inputconfig/impl/InputConfigImpl.java [hunk 2]
```diff
@@ -39,15 +39,17 @@
...
   )
   @Expose
   private List<InputDescriptorImpl> input;
 
   @ShipperConfigElementDescription(
     path = "/filter",
     type = "list of json objects",
-    description = "A list of filter descriptions"
+    description = "A list of filter descriptions",
+    examples = {"{\"filter\" : [ {\"filter\": \"json\", \"conditions\": {\"fields\": { \"type\": [\"mytype1\", \"mytype2\"]} } } ]}"}
```
Comment (kasakrisz) [inline ambari-logsearch-config-json/src/main/java/org/apache/ambari/logsearch/config/json/model/inputconfig/impl/InputConfigImpl.java]: Same as above (line 43). Giving a concrete example could be useful.
```
  "filter":[
    {
      "filter":"json",
      "conditions":{
        "fields":{
          "type":[
            "logsearch_app",
            "logsearch_feeder",
            "logsearch_perf"
          ]
        }
      }
    }
  ]

```

File: ambari-logsearch-config-json/src/main/java/org/apache/ambari/logsearch/config/json/model/inputconfig/impl/InputS3FileDescriptorImpl.java [hunk 3]
```diff
@@ -34,7 +34,7 @@
   @ShipperConfigElementDescription(
     path = "/input/[]/s3_access_key",
     type = "string",
-    description = "The access key used for AWS credentials."
+    description = "The access key used for AWS credentials. (Not supported yet through shipper configurations)"
```
Comment (kasakrisz) [inline ambari-logsearch-config-json/src/main/java/org/apache/ambari/logsearch/config/json/model/inputconfig/impl/InputS3FileDescriptorImpl.java]: Is it possible to omit these not supported properties from the doc? I think these should only be included if the functionality is supported.
lines: 37, 46, 55

File: ambari-logsearch-docs/src/main/java/org/apache/ambari/logsearch/doc/LogSearchDocumentationGenerator.java [hunk 4]
```diff
@@ -130,12 +140,32 @@ public static void main(String[] args) {
...
+    beanConfig.configure(swagger);
+    beanConfig.scanAndRead();
+    String yaml = Yaml.mapper().writeValueAsString(swagger);
+    StringBuilder b = new StringBuilder();
+    String[] parts = yaml.split("\n");
+    for (String part : parts) {
+      b.append(part);
+      b.append("\n");
+    }
+    return b.toString();
```
Comment (kasakrisz) [inline ambari-logsearch-docs/src/main/java/org/apache/ambari/logsearch/doc/LogSearchDocumentationGenerator.java]: What is the difference between the result of `b.toString()` and `yaml` ?

File: ambari-logsearch-docs/src/main/resources/templates/shipper_configurations.md.ftl [hunk 5]
```diff
@@ -17,19 +17,117 @@ limitations under the License.
...
 
+Input configurations (for monitoring logs) can be defined in the input descriptor section.
+Example:
+```json
+{
+  "input" : [
+    {
+      "type": "simple_service",
+      "rowtype": "service",
+      "path": "/var/logs/my/service/service_sample.txt",
```
Comment (kasakrisz) [inline ambari-logsearch-docs/src/main/resources/templates/shipper_configurations.md.ftl]: Maybe `service_sample.log` would be more useful.

File: ambari-logsearch-docs/src/main/resources/templates/shipper_configurations.md.ftl [hunk 6]
```diff
@@ -38,8 +136,97 @@ limitations under the License.
...
+          "type":[
+            "simple_service",
+            "simple_audit_service",
+            "docker_service"
+          ]
+        }
+      },
+      "log4j_format":"%d{ISO8601} %5p [%t] %c{1}:%L - %m%n",
+      "multiline_pattern":"^(%{TIMESTAMP_ISO8601:logtime})",
+      "message_pattern":"(?m)^%{TIMESTAMP_ISO8601:logtime}%{SPACE}%{LOGLEVEL:level}%{SPACE}\\[%{DATA:thread_name}\\]%{SPACE}%{JAVACLASS:logger_name}:%{INT:line_number}%{SPACE}-%{SPACE}%{GREEDYDATA:log_message}",
```
Comment (kasakrisz) [inline ambari-logsearch-docs/src/main/resources/templates/shipper_configurations.md.ftl]: Including some lines of sample log can be useful to understand what log entries can be parsed with this pattern.

File: ambari-logsearch-config-json/src/main/java/org/apache/ambari/logsearch/config/json/model/inputconfig/impl/InputConfigImpl.java (hunk 1 above)
Comment (oleewere) [reply ambari-logsearch-config-json/src/main/java/org/apache/ambari/logsearch/config/json/model/inputconfig/impl/InputConfigImpl.java]: @kasakrisz ok, could be

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/109/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/110/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/111/
Test PASSed.

**Pull request merged**